cmd/argo-watcher/argo_status_updater.go

```diff
@@ -76,6 +78,20 @@ func (updater *ArgoStatusUpdater) waitForApplicationDeployment(task models.Task)
 	var application *models.Application
 	var err error
 
+	app, err := updater.argo.api.GetApplication(task.App)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Why do we need this call here? Should it be done when we start updating git image tag (e.g. in the IF condition down the line) ?

Reply:
We need to get application details to collect its annotations. In the if block, we already making decisions based on those annotations.